src/reporters/udp_sender.js

```diff
@@ -132,19 +135,23 @@ export default class UDPSender {
 
         if (writeResult.err) {
             this._logger.error(`error writing Thrift object: ${writeResult.err}`);
-            return {err: true, numSpans: numSpans};
+            if (callback) {
+              callback({err: true, numSpans: numSpans});
+            }
+            return;
         }
 
         // Having the error callback here does not prevent uncaught exception from being thrown,
         // that's why in the constructor we also add a general on('error') handler.
-        this._client.send(thriftBuffer, 0, thriftBuffer.length, this._port, this._host, (err, sent) => {
+        this._client.send(thriftBuffer, 0, thriftBuffer.length, this._port, this._host, (err) => {
             if (err) {
                 this._logger.error(`error sending spans over UDP: ${err}, packet size: ${writeResult.offset}, bytes sent: ${sent}`);
             }
+            if (callback) {
+              callback({err: err !== 0, numSpans: numSpans});
+            }
         });
         this._reset();
```

Comment:
this doesn't look right here.

Reply:
I had been thinking that the error that was passed into the optional callback (which is being passed in here) was a number: zero representing essentially no errors, other numbers denoting some type of error.  But this error that is being passed to the callback, is it in fact an object?
I've been looking for a definition of the error. [This test in node repo](https://github.com/nodejs/node/blob/98e54b0bd4854bdb3e2949d1b6b20d6777fb7cde/test/parallel/test-dgram-msgsize.js) implies that it is an error object.

Reply:
I was referring to the call to reset(). I think that logic is altogether broken since there are no guarantees that more spans won't be added to the buffer while the send is in progress.

Cc @black-adder